bot/lords_and_knights_v3.rb

```diff
@@ -96,44 +97,45 @@ def choose_tab(title)
 
     def choose_building(titles)
       titles = [titles] unless titles.is_a?(Array)
-      logger.debug(": choose_building #{titles.join(', ')}")
-      result = nil
-      available_buildings = all('#menu-section-general-container > .menu-section > .menu--content-section .menu-list-element.menu-list-element-basic.clickable.with-icon-left.with-icon-right')
+      
+      logger.debug "   * choose_building #{titles.join(', ')}"
+      available_buildings = all('.menu-list-element.menu-list-element-basic.clickable.with-icon-left.with-icon-right') 
+      
       available_buildings.each do |building|
         building_name = building.first('.menu-list-element-basic--title').text()
         next unless titles.include?(building_name)
-        result = building
+        logger.debug "   * click building #{building_name}"
         building.click()
         timeout
         wait_while '#over-layer--game-pending.in'
-        break
+        return true
       end
-      result
+      return false
     end
 
     def choose_building_list
-      logger.debug(": choose_building_list")
+      logger.debug("   * choose_building_list")
       return if has_selector?('div.top-bar-button--HabitatBuildings.active')
       first('div.top-bar-button--HabitatBuildings').click()
     end
 
     def choose_tavern
       choose_building_list
-      choose_building(["Tavern", "Tavern Quarter"])
+      return choose_building(["Tavern", "Tavern Quarter"])
```

Comment:
I see you want to have `return` in each method. It is not a Ruby style, but I am ok with it.

In this case better to `return true` so it garantees next usage base on Success or Error instead of Object.

Reply:
There are in other places `choose_*` that should have similar behavior.